src/AddressRegistry/StreetName/StreetNameState.cs

```diff
@@ -36,28 +36,29 @@ private StreetName()
 
             Register<AddressWasMigratedToStreetName>(When);
             Register<AddressWasProposedV2>(When);
-            Register<AddressWasApproved>(When);
-            Register<AddressWasCorrectedFromApprovedToProposed>(When);
-            Register<AddressWasCorrectedFromApprovedToProposedBecauseHouseNumberWasCorrected>(When);
-            Register<AddressWasRejected>(When);
-            Register<AddressWasRejectedBecauseHouseNumberWasRejected>(When);
-            Register<AddressWasRejectedBecauseHouseNumberWasRetired>(When);
-            Register<AddressWasRejectedBecauseStreetNameWasRetired>(When);
-            Register<AddressWasDeregulated>(When);
-            Register<AddressWasRegularized>(When);
-            Register<AddressWasRetiredV2>(When);
-            Register<AddressWasRetiredBecauseHouseNumberWasRetired>(When);
-            Register<AddressWasRetiredBecauseStreetNameWasRetired>(When);
-            Register<AddressWasCorrectedFromRetiredToCurrent>(When);
-            Register<AddressPositionWasChanged>(When);
-            Register<AddressPostalCodeWasChangedV2>(When);
-            Register<AddressPositionWasCorrectedV2>(When);
-            Register<AddressPostalCodeWasCorrectedV2>(When);
-            Register<AddressHouseNumberWasCorrectedV2>(When);
-            Register<AddressBoxNumberWasCorrectedV2>(When);
-            Register<AddressWasRemovedV2>(When);
-            Register<AddressWasCorrectedFromRejectedToProposed>(When);
-            Register<AddressWasRemovedBecauseHouseNumberWasRemoved>(When);
+            Register<AddressWasApproved>(RouteEventToRelevantAddress);
```

Comment:
Can you change back to `When` ? It's consistent with rest of registries and handles When the event occurs.